ce/api/geo.py

```diff
@@ -52,88 +53,144 @@ def inner(obj, _seen_ids = None):
+        #performance, but these aren't used in production and can generally be ignored.
+        class CachedFunction(object):
+            def __init__(self, maxsize, keyfunc, func):
+                self.hits = self.misses = 0
+                self.maxsize = maxsize
+                self.keyfunc = keyfunc
+                self.func = func
+                self.func.__name__ = func.__name__
+                self.cache = OrderedDict()
+                self.MBconversion = 1024 * 1024
+
+
+            def __call__(self, *args):
+                key = self.keyfunc(*args)
+                log.debug('Checking cache for key {}'.format(key))
+
+                with cache_lock:
+                    try:
+                        result = self.cache[key]
+                        self.cache.move_to_end(key)
+                        self.hits += 1
+                        log.debug("Cache Hit for {}".format(self.func))
+                        return result
+                    except KeyError:
+                        pass
+
+                    log.debug('Cache MISS for {}'.format(self.func))
+                    result = self.func(*args)
+
+                with cache_lock:
+                    self.cache[key] = result
+                    self.misses += 1
+                    while getsize(self.cache) > self.maxsize * self.MBconversion:
+                        if len(self.cache) == 1:
+                            log.warning("Cache maxsize is set to {} MB ".format(self.maxsize),
+                                        "but tried to cache a {} MB item".format(getsize(self.cache) / self.MBconversion))
+                        self.cache.popitem(0)
 
-        with cache_lock:
-            self.cache[key] = result
-            self.misses += 1
-            while getsize(self.cache) > self.maxsize * 1024 * 1024: # convert to MB
-                self.cache.popitem(0) # Purge least recently used cache entry
+                return result
 
-        return result
+            #cache testing / debugging functions
+            def cache_clear(self):
+                with cache_lock:
+                    self.cache.clear()
+                    self.hits = 0
+                    self.misses = 0
 
-    def cache_clear(self):
-        with cache_lock:
-            self.cache.clear()
-            self.hits = 0
-            self.misses = 0
+            def get_hits(self): return self.hits
 
-@memoize_mask
-def wkt_to_masked_array(nc, fname, wkt, variable):
-    poly = loads(wkt)
-    return polygon_to_masked_array(nc, fname, poly, variable)
+            def get_misses(self): return self.misses
 
+            def get_size(self): return getsize(self.cache)
 
-def polygon_to_masked_array(nc, fname, poly, variable):
+            def get_length(self): return len(self.cache)
+
+        cf = CachedFunction(self.maxsize, self.keyfunc, func)
+        return cf
 
+def make_mask_grid_key(nc, fname, poly, variable):
```

Comment:
Yeah, this is kind of a pain to pull all of the grid information out, so good job on stashing it all away in it's own function.